Signal/src/ViewControllers/SelectRecipientViewController.m

```diff
@@ -0,0 +1,553 @@
+{
+    UIView *row = [UIView new];
+    [superview addSubview:row];
+    [row autoPinWidthToSuperview];
+    if (previousRow) {
+        [row autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:previousRow withOffset:0];
+    } else {
+        [row autoPinEdgeToSuperviewEdge:ALEdgeTop];
+    }
+    [row autoSetDimension:ALDimensionHeight toSize:height];
+    return row;
+}
+
+#pragma mark - Country
+
+- (void)populateDefaultCountryNameAndCode
+{
+    NSLocale *locale = NSLocale.currentLocale;
+    NSString *countryCode = [locale objectForKey:NSLocaleCountryCode];
+    NSNumber *callingCode = [[PhoneNumberUtil sharedUtil].nbPhoneNumberUtil getCountryCodeForRegion:countryCode];
+    NSString *countryName = [PhoneNumberUtil countryNameFromCountryCode:countryCode];
+    [self updateCountryWithName:countryName
+                    callingCode:[NSString stringWithFormat:@"%@%@", COUNTRY_CODE_PREFIX, callingCode]
+                    countryCode:countryCode];
+}
+
+- (void)updateCountryWithName:(NSString *)countryName
+                  callingCode:(NSString *)callingCode
+                  countryCode:(NSString *)countryCode
+{
+
+    _callingCode = callingCode;
+
+    NSString *title = [NSString stringWithFormat:@"%@ (%@)", callingCode, countryCode.uppercaseString];
+    [self.countryCodeButton setTitle:title forState:UIControlStateNormal];
+    [self.countryCodeButton layoutSubviews];
+}
+
+- (void)setCallingCode:(NSString *)callingCode
+{
+    _callingCode = callingCode;
+
+    [self updatePhoneNumberButtonEnabling];
+}
+
+#pragma mark - Actions
+
+- (void)showCountryCodeView:(nullable id)sender
+{
+    CountryCodeViewController *countryCodeController = [[UIStoryboard storyboardWithName:@"Registration" bundle:NULL]
+        instantiateViewControllerWithIdentifier:@"CountryCodeViewController"];
+    countryCodeController.delegate = self;
+    countryCodeController.shouldDismissWithoutSegue = YES;
+    UINavigationController *navigationController =
+        [[UINavigationController alloc] initWithRootViewController:countryCodeController];
+    [self presentViewController:navigationController animated:YES completion:[UIUtil modalCompletionBlock]];
+}
+
+- (void)phoneNumberButtonPressed:(id)sender
+{
+    [self tryToSelectPhoneNumber];
+}
+
+- (void)tryToSelectPhoneNumber
+{
+    OWSAssert(self.delegate);
+
+    if (![self hasValidPhoneNumber]) {
+        OWSAssert(0);
+        return;
+    }
+
+    NSString *rawPhoneNumber = [self.callingCode stringByAppendingString:self.phoneNumberTextField.text.digitsOnly];
+
+    NSMutableArray<NSString *> *possiblePhoneNumbers = [NSMutableArray new];
+    for (PhoneNumber *phoneNumber in
+        [PhoneNumber tryParsePhoneNumbersFromsUserSpecifiedText:rawPhoneNumber
+                                              clientPhoneNumber:[TSAccountManager localNumber]]) {
+        [possiblePhoneNumbers addObject:phoneNumber.toE164];
+    }
+    if ([possiblePhoneNumbers count] < 1) {
+        OWSAssert(0);
```

Comment:
Ditto added logic. This should never happen since the button should be disabled.